src/main/java/io/gravitee/policy/jsonvalidation/JsonValidationPolicy.java

```diff
@@ -61,53 +69,142 @@ public JsonValidationPolicy(JsonValidationPolicyConfiguration configuration) {
+                                if (!configuration.isStraightRespondMod()) {
+                                    new ErrorResponse()
+                                            .policyChain(policyChain)
+                                            .executionContext(executionContext)
+                                            .key(JSON_INVALID_RESPONSE_PAYLOAD_KEY)
+                                            .errorMessage(INTERNAL_ERROR)
+                                            .httpStatusCode(HttpStatusCode.INTERNAL_SERVER_ERROR_500)
+                                            .send();
+                                } else {
+                                    return Buffer.buffer(buffer.toString());
+                                }
+                            } else {
+                                return Buffer.buffer(buffer.toString());
+                            }
+
+                        } catch (Exception ex) {
+                            request.metrics().setMessage(ex.toString());
+                            if (!configuration.isStraightRespondMod()) {
+                                new ErrorResponse()
+                                        .policyChain(policyChain)
+                                        .executionContext(executionContext)
+                                        .key(JSON_INVALID_RESPONSE_FORMAT_KEY)
+                                        .errorMessage(INTERNAL_ERROR)
+                                        .httpStatusCode(HttpStatusCode.INTERNAL_SERVER_ERROR_500)
+                                        .send();
+                            } else {
+                                return Buffer.buffer(buffer.toString());
+                            }
+                        }
+                        return Buffer.buffer(buffer.toString());
+                    }).build();
+        }
+        return null;
     }
 
-    private void sendBadRequestResponse(String key, ExecutionContext executionContext, PolicyChain policyChain) {
-        if (configuration.getErrorMessage() != null && !configuration.getErrorMessage().isEmpty()) {
-            String errorMessage = executionContext.getTemplateEngine().convert(configuration.getErrorMessage());
-            policyChain.streamFailWith(PolicyResult.failure(key, HttpStatusCode.BAD_REQUEST_400, errorMessage, MediaType.APPLICATION_JSON));
+    private ProcessingReport getReport(JsonNode schema, JsonNode content) throws ProcessingException {
+        if (configuration.isValidateUnchecked()) {
+            return validator.validateUnchecked(schema, content, configuration.isDeepCheck());
         } else {
-            String errorMessage = "Bad request";
-            policyChain.streamFailWith(PolicyResult.failure(key, HttpStatusCode.BAD_REQUEST_400, errorMessage, MediaType.TEXT_PLAIN));
+            return validator.validate(schema, content, configuration.isDeepCheck());
+        }
+    }
+
+    private class ErrorResponse {
```

Comment:
I do not see the added value of this inner class against `policyChain.streamFailWith();`.
Can you enlighten me?

Reply:
With adding `httpStatusCode` parameter to method sendResponse, it will have 5 input arguments. To my mind, that`s just not very good idea to have such multi-args methods. But I can revert that if you want, there is no problem.

Reply:
Which `sendResponse` are you talking about?

Reply:
`sendBadRequestResponse`